src/redux/books/books.js

```diff
@@ -25,18 +12,50 @@ export default function booksReducer(state = initialState, action = {}) {
     case ADD_BOOK:
       return [...state, action.payload];
     case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.payload);
+      return state.filter((book) => book.item_id !== action.payload);
+    case GET_BOOKS:
+      return action.payload;
     default:
       return state;
   }
 }
 // Action creators
-const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
-});
-const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  payload: id,
-});
-export { addBook, removeBook };
+function getBooks() {
+  return async (dispatch) => {
+    const response = await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/rkZUelgZDiisQq3YPSnE/books');
+    const data = await response.json();
+    const books = [];
+    Object.keys(data).forEach((e) => {
+      books.push({ ...data[e][0], item_id: e });
+    });
+    dispatch({
+      type: GET_BOOKS,
+      payload: books,
+    });
+  };
+}
+
+const addBook = (book) => async (dispatch) => {
+  await fetch(API_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(book),
+  });
+  dispatch({
+    type: ADD_BOOK,
+    payload: book,
+  });
+};
+
```

Comment:
- [ ] Its great that you have implemented the changes regarding the remove book functionality :+1: but i apologize on behalf of previous reviewer because he didn't mention that your add book api call is not working properly. When i add an book it doesn't go to the api. Kindly check it and fix it :+1: